src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -958,4 +965,46 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
             }
         }
     }
+
+    // WA for OneDNN binary add fusions: we need to broadcast batch dimension to avoid situation with
+    // batch dimension mismatch in OneDNN tensor descriptors as follow:
+    // * Gemm output shape: (b,f,y,x) -> OneDNN shape: (b*f,y,x)
+    // * Gemm fused op shape: (1,f,y,x) -> OneDNN shape: (1*f,y,x)
+    // If batch dimension of gemm output is not equal to 1, then OneDNN will not be able to broadcast fused op data
+    // correctly and we need to do it manually
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    for (auto& node : p.get_processing_order()) {
+        if (node->is_type<gemm>() && node->get_preferred_impl_type() == impl_types::onednn) {
+            for (const auto& fused_prim : node->get_fused_primitives()) {
+                if (fused_prim.is_type<eltwise>() &&
+                    one_of(fused_prim.typed_desc<eltwise>()->mode, eltwise_mode::sum, eltwise_mode::sub, eltwise_mode::prod)) {
+                    auto& data = node->get_dependency(fused_prim.dep_start_idx);
+
+                    auto gemm_layout = node->get_output_layout();
+                    auto gemm_dims = onednn::convert_gemm_tensor(gemm_layout.get_tensor(),
+                                                                 cldnn::format::dimension(gemm_layout.format),
+                                                                 false);
+
+                    auto data_layout = data.get_output_layout();
+                    auto data_dims = onednn::convert_gemm_tensor(data_layout.get_tensor(),
+                                                                 cldnn::format::dimension(data_layout.format),
+                                                                 false);
+
+                    if (gemm_dims[0] == data_dims[0])
+                        continue;
+
+                    auto broadcast_size = data_layout.get_shape();
+                    broadcast_size[0] = gemm_dims[0] / broadcast_size[1];
```

Comment:
@sshlyapn 
we have two cases of broadcasting: `[b,1,y,x] -> [b,f,y,x]` and `[1,f,y,x] -> [b,f,y,x]`
I think it is handling only one of these two.
What about getting `b` and `f` sizes from `gemm_layout`?

Reply:
I haven't seen such cases, but yes, it looks like we can handle second case too with using `gemm_layout.get_shape()` as `broadcast_size`

Reply:
Applied